### Upload the 2017-2018 school year file "MERGED2017_18_PP.csv" to Colab/Deepnote or any Cloud environment of your choice (you can use your local environment too) and create a Jupyter notebook to explore the dataset:

In [1]:
import pandas as pd
df = pd.read_csv('MERGED2017_18_PP.csv')

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2718: DtypeWarning: Columns (1729) have mixed types.Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


### how many colleges or rows?
### how many variables or columns?

In [3]:
df.shape

(7112, 1986)

In [4]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7112 entries, 0 to 7111
Columns: 1986 entries, UNITID to SCUGFFN_POOLED
dtypes: float64(1931), int64(12), object(43)
memory usage: 107.8+ MB


**There are 7112 rows(colleges) and 1986 colums(variables) in this dataset**

### Look at the first 5 colleges

In [8]:
df.head(5)

UNITID    OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
0  100654   100200  ...                       NaN            NaN
1  100663   105200  ...                       NaN            NaN
2  100690  2503400  ...                       NaN            NaN
3  100706   105500  ...                       NaN            NaN
4  100724   100500  ...                       NaN            NaN

[5 rows x 1986 columns]

### Look at the last 5 colleges

In [9]:
df.tail(5)

UNITID    OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
7107  48312404   108163  ...                       NaN            NaN
7108  48387801  4223701  ...                       NaN            NaN
7109  48400201  4228101  ...                       NaN            NaN
7110  48778201  4221501  ...                       NaN            NaN
7111  49005401  4182601  ...                       NaN            NaN

[5 rows x 1986 columns]

### Look at a random sample of 5 colleges

In [10]:
df.sample(5)

UNITID    OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
5539  456029  3253300  ...                       NaN            NaN
5814  461148  4118800  ...                       NaN            NaN
5173  447403   822105  ...                       NaN            NaN
2234  190169   678900  ...                       NaN            NaN
4869  440004  3437300  ...                       NaN            NaN

[5 rows x 1986 columns]

### 7. Since there are too many variables to explore, many of which are not interesting. Create a smaller data frame with the following 9 variables:

In [11]:
smaller_df = df[["UNITID","INSTNM","CITY","STABBR","ZIP","ADM_RATE","UGDS","TUITIONFEE_IN","MN_EARN_WNE_P10"]]
smaller_df

UNITID  ... MN_EARN_WNE_P10
0       100654  ...             NaN
1       100663  ...             NaN
2       100690  ...             NaN
3       100706  ...             NaN
4       100724  ...             NaN
...        ...  ...             ...
7107  48312404  ...             NaN
7108  48387801  ...             NaN
7109  48400201  ...             NaN
7110  48778201  ...             NaN
7111  49005401  ...             NaN

[7112 rows x 9 columns]

### 8. Download and read (use naked eyes, no coding) the data dictionary file - https://collegescorecard.ed.gov/assets/CollegeScorecardDataDictionary.xlsx
### 9. Find out the definition of the above 9 variables and document the definitions in the Jupyter Notebook in a Markdown cell so that the readers of your notebook will know the meaning of these variables.

1. UNITID - Unit ID for Institution
2. INSTNM - Institution Name
3. CITY - City Institution is located in
4. STABBR - State Institution is located in
5. ZIP - Zip Code of Institution
6. ADM_RATE - Admission Rate
7. UGDS - Enrollment of undergraduate certificate/degree-seeking students
8. TUITIONFEE_IN - In-state tuition and fees
9. MN_EARN_WNE_P10 - Mean earnings of students working and not enrolled 10 years after entry

### 10.Explore the smaller data frame

### Look at the first 10 colleges

In [12]:
smaller_df.head(10)

UNITID                               INSTNM  ... TUITIONFEE_IN MN_EARN_WNE_P10
0  100654             Alabama A & M University  ...        9857.0             NaN
1  100663  University of Alabama at Birmingham  ...        8328.0             NaN
2  100690                   Amridge University  ...        6900.0             NaN
3  100706  University of Alabama in Huntsville  ...       10280.0             NaN
4  100724             Alabama State University  ...       11068.0             NaN
5  100751            The University of Alabama  ...       10780.0             NaN
6  100760    Central Alabama Community College  ...        4440.0             NaN
7  100812              Athens State University  ...           NaN             NaN
8  100830      Auburn University at Montgomery  ...        8020.0             NaN
9  100858                    Auburn University  ...       10968.0             NaN

[10 rows x 9 columns]

### Look at the last 10 colleges

In [13]:
smaller_df.tail(10)

UNITID  ... MN_EARN_WNE_P10
7102  48154401  ...             NaN
7103  48305401  ...             NaN
7104  48312401  ...             NaN
7105  48312402  ...             NaN
7106  48312403  ...             NaN
7107  48312404  ...             NaN
7108  48387801  ...             NaN
7109  48400201  ...             NaN
7110  48778201  ...             NaN
7111  49005401  ...             NaN

[10 rows x 9 columns]

### Look at a random sample of 10 colleges

In [14]:
smaller_df.sample(10)

UNITID  ... MN_EARN_WNE_P10
6401  486248  ...             NaN
254   111799  ...             NaN
4924  441496  ...             NaN
4331  383279  ...             NaN
1286  156134  ...             NaN
1809  175005  ...             NaN
5519  455798  ...             NaN
5039  444228  ...             NaN
2802  206349  ...             NaN
3891  240480  ...             NaN

[10 rows x 9 columns]

### Find and display UMBC

In [16]:
df[df['CITY']=='Baltimore']

UNITID    OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
1451    161855  2494700  ...                       NaN            NaN
1452    161864   206100  ...                       NaN            NaN
1453    161873   210200  ...                       NaN            NaN
1460    162283   206800  ...                       NaN            NaN
1462    162371  2545400  ...                       NaN            NaN
1466    162654   207300  ...                       NaN            NaN
1473    162928   207700  ...                       NaN            NaN
1475    163046   207800  ...                       NaN            NaN
1477    163259   210400  ...                       NaN            NaN
1478    163268   210500  ...                       NaN            NaN
1480    163295   208000  ...                       NaN            NaN
1485    163453   208300  ...                       NaN            NaN
1487    163532   208700  ...                       NaN            NaN
1489    163578   206500  ...                       NaN            NaN
1491    163736   749100  ...                       NaN            NaN
1493    163815  2331000  ...                       NaN            NaN
3004    212452  3667300  ...                       NaN            NaN
4225    373784  3493300  ...                       NaN            NaN
4762    434672   206300  ...                       NaN            NaN
5314    450076  3493301  ...                       NaN            NaN
5834    461379  4168900  ...                       NaN            NaN
6016    476601  4188400  ...                       NaN            NaN
7006  43018403   145924  ...                       NaN            NaN

[23 rows x 1986 columns]

In [17]:
df[df['INSTNM']=='University of Maryland-Baltimore County']

UNITID   OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
1478  163268  210500  ...                       NaN            NaN

[1 rows x 1986 columns]

### Find Univ. of Maryland College Park

In [18]:
df[df['INSTNM']=='University of Maryland-College Park']

UNITID   OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
1479  163286  210300  ...                       NaN            NaN

[1 rows x 1986 columns]

### Find and display Loyola College Maryland

In [19]:
df[df['INSTNM']=='Loyola University Maryland']

UNITID   OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
1475  163046  207800  ...                       NaN            NaN

[1 rows x 1986 columns]

### Find and display Johns Hopkins

In [20]:
df[df['INSTNM']=='Johns Hopkins University']

UNITID   OPEID  ...  FTFTPCTFLOAN_POOLED_SUPP SCUGFFN_POOLED
1473  162928  207700  ...                       NaN            NaN

[1 rows x 1986 columns]

### Calculate the min, max, and average in-state tution of the above four institutions.

In [29]:
my_df = ['University of Maryland-Baltimore County','University of Maryland-College Park','Loyola University Maryland','Johns Hopkins University']

smaller_df[smaller_df.INSTNM.isin(my_df)] #only extracts the info for the 4 colleges listed

UNITID  ... MN_EARN_WNE_P10
1473  162928  ...             NaN
1475  163046  ...             NaN
1478  163268  ...             NaN
1479  163286  ...             NaN

[4 rows x 9 columns]

In [30]:
Tuition = [52170.0,47560.0,11518.0,10399.0]
Tuition

[52170.0, 47560.0, 11518.0, 10399.0]

**Minimum**

In [31]:
min(Tuition)

10399.0

**Maximum**

In [32]:
max(Tuition)

52170.0

**Average**

In [33]:
Avg_Tuition = sum(Tuition) / len(Tuition)
Avg_Tuition

30411.75

### 11.Filter the smaller dataset further to only keep colleges from Maryland, Virgina, Washington DC, Delaware, and Pennsylvinia. Save the filtered dataset to a new file named "Five_States_Colleges.csv" for future analysis. 

In [34]:
states_5 = ["MD","VA","DC","DE","PA"] # this will extract all colleges in these states

states_df = smaller_df.loc[smaller_df['STABBR'].isin(states_5)]

states_df

UNITID  ... MN_EARN_WNE_P10
634     130837  ...             NaN
635     130873  ...             NaN
636     130907  ...             NaN
637     130934  ...             NaN
638     130943  ...             NaN
...        ...  ...             ...
7063  45794301  ...             NaN
7084  47503303  ...             NaN
7089  47503308  ...             NaN
7094  47503313  ...             NaN
7103  48305401  ...             NaN

[690 rows x 9 columns]

In [35]:
states_df.to_csv(r'Five_States_Colleges.csv', index = False)